index.js

```diff
@@ -48,55 +48,37 @@ const loginBtnsEl = document.getElementById("login-btns")
 const logoutBtnEl = document.getElementById("logout-btn")
 const signUpBtn = document.getElementById("sign-up-btn")
 const signInBtn = document.getElementById("sign-in-btn")
-const xOutModalBtnsArr = document.querySelectorAll(".x-out")
 const buttonsArr = document.querySelector("#container").querySelectorAll(".btn")
 const createUserBtn = document.getElementById("create-user")
 const loginBtn = document.getElementById("login-btn")
 const logoutBtn = document.getElementById("log-out-btn")
 const signUpForm = document.getElementById("sign-up-form")
 const signInForm = document.getElementById("sign-in-form")
+const signUpWrapper = document.getElementById("sign-up-wrapper")
+const signInWrapper = document.getElementById("sign-in-wrapper")
 let isValid = false
 
-document.querySelectorAll(".login-btn").forEach(function(loginBtn) {
-    loginBtn.addEventListener("click", function() {
-        displayGenericModal(`${loginBtn.dataset.modalType}`)
-    })
-})
-
-signUpBtn.addEventListener("click", function() {
-    displayGenericModal("sign-up")
-})
-
-signInBtn.addEventListener("click", function() {
-    displayGenericModal("sign-in")
-})
-
-xOutModalBtnsArr.forEach(function(button) {
-    button.addEventListener("click", function(event) {
-        const currentModal = event.target.parentElement
-
-        hideModal(currentModal)
-
-        document.querySelectorAll(".login-btn").forEach(function(loginBtn) {
-            enableBtn(loginBtn)
-        })
-    })
+document.addEventListener("click", event => {
+    [...event.target.classList].includes('x-out') ? event.target.parentElement.parentElement.classList.toggle('hidden') :
+    event.target === signUpBtn ? displayGenericModal('sign-up') :
+    event.target === signUpWrapper ? displayGenericModal('sign-up') : 
+    event.target === signInBtn ? displayGenericModal('sign-in') : 
+    event.target === signInWrapper ? displayGenericModal('sign-in') :
+    [...event.target.classList].includes('submit-btn') ? event.target.parentElement.parentElement.classList.toggle('hidden') : ''
 })
 
 createUserBtn.addEventListener("click", function(event) {
-    const signUpModal = event.target.parentElement.parentElement
-
     createUser()
     enableBtns()
-    hideModal(signUpModal)
+    displayGenericModal('sign-up')
     toggleLogInOutBtns()
 
     loggedIn = true
 })
 
-loginBtn.addEventListener("click", function(event) {
+loginBtn.addEventListener("click", function() {
     enableBtns()
-    hideModal(event.target.parentElement.parentElement)
+    displayGenericModal('sign-in')
```

Comment:
This is an issue that I think I introduced but we should make sign-in and log-in one or the other